### ISSUE TITLE
Custom condition for evolve

### ISSUE THREAD

**Issue by HenrikMettler** [Happy-Algorithms-League/hal-cgp#324]
Currently the `while`loop in evolve works with a maximal number of generations or a maximal number of calls of the objective function (with a break if termination fitness is reached). However other conditions might be desirable (eg. a max wall clock-time). 

For maximum flexibility we could evaluate whether to continue or not, with a custom function: 
```python
def continue_run(pop, ea)
    return _criterium_

# in hl_api.py
while continue_run(pop, ea)
    # ...
```

Probably we would need some predefined functions for better user experience, eg: 
```python
def max_generations(max_generations)
    def continue_run(pop, ea)
          return pop.generation < max_generations
    return continue_run
```

**Comment by jakobj** [comment]
good idea :) we should keep in mind use cases where more than one exit criterion is desired, for example currently we exit upon reaching a max number of generations _or_ a certain fitness value. how about allowing both a function and a list of functions to be passed to the `hl_api.evolve` function? internally we could then either evaluate just a the single function or evaluate all functions in the list and make sure the are all true.
